### PR TITLE
Fix #942

### LINKED ISSUE #942: Using setNotifyAboutChanges appears to ignore some entity meta
[Video for example](https://cdn.discordapp.com/attachments/706185253974442006/965626658323300432/2022-04-18_15-51-40.mp4)
[Code to reproduce](https://bin.bloom.host/uniroyesuz.swift)

In the video you can see when I type reset2 (the one without `setNotifyAboutChanges`) everything works fine, but when I do reset it does not send certain information at all (e.g. setGravity) or delays them until another action (e.g. setInvisible seemed to fix itself after I crouched)

### PR DESCRIPTION
**Pull request by WeiiswurstDev**
The bug was caused because the list of changes was cleared before it was sent to the players. It will now get cleared after it is sent.

### CODE REVIEW

**Comment by WeiiswurstDev** [comment]
I tested the fix locally and the code for reproducing the bug does not reproduce it anymore. Feel free to test it yourself with the code in the original issue. For my testing, I simply overwrote the main method in the example (however, this is not part of the commit / the PR) with the test code.

Therefore, this PR fixes #942

**Comment by TheMode** [comment]
Could you write a test case for it?

**Comment by WeiiswurstDev** [comment]
Ok, will push it soon™

File: src/test/java/net/minestom/server/entity/EntityMetaTest.java [hunk 1]
```diff
@@ -21,4 +30,76 @@ public void ensureRegistration() throws IllegalAccessException {
         }
         assertTrue(list.isEmpty(), "Missing meta for: " + list);
     }
+
+
+    @Test
+    public void notifyAboutChanges(Env env) {
```
Comment (TheMode) [inline src/test/java/net/minestom/server/entity/EntityMetaTest.java:36]: This should be put in a dedicated `EntityMetaIntegrationTest` file (and allow the changes above to be reverted)

Comment (WeiiswurstDev) [reply src/test/java/net/minestom/server/entity/EntityMetaTest.java:36]: Alright, will move it into a new test

Comment (WeiiswurstDev) [reply src/test/java/net/minestom/server/entity/EntityMetaTest.java:36]: Should I amend the commit?

Comment (TheMode) [reply src/test/java/net/minestom/server/entity/EntityMetaTest.java:36]: branch are squashed before merge anyway. so don't worry about the git history

Comment (WeiiswurstDev) [reply src/test/java/net/minestom/server/entity/EntityMetaTest.java]: ok

File: src/test/java/net/minestom/server/entity/EntityMetaTest.java [hunk 2]
```diff
@@ -1,11 +1,18 @@
...
+import net.minestom.server.network.packet.server.play.EntityMetaDataPacket;
+import org.junit.jupiter.api.Assertions;
 import org.junit.jupiter.api.Test;
 
 import java.util.ArrayList;
 import java.util.List;
+import java.util.function.Consumer;
 
-import static org.junit.jupiter.api.Assertions.assertTrue;
+import static org.junit.jupiter.api.Assertions.*;
```
Comment (TheMode) [inline src/test/java/net/minestom/server/entity/EntityMetaTest.java:15]: Can you remove the line changes from this file?

Comment (TheMode) [reply src/test/java/net/minestom/server/entity/EntityMetaTest.java]: (There are still multiple line changes)

Comment (WeiiswurstDev) [reply src/test/java/net/minestom/server/entity/EntityMetaTest.java]: ok

File: src/test/java/net/minestom/server/entity/EntityMetaTest.java [hunk 3]
```diff
@@ -21,4 +21,4 @@ public void ensureRegistration() throws IllegalAccessException {
         }
         assertTrue(list.isEmpty(), "Missing meta for: " + list);
     }
-}
+}
```
Comment (TheMode) [inline src/test/java/net/minestom/server/entity/EntityMetaTest.java:24]: Still a line change

**Comment by TheMode** [comment]
Thanks for the fix & awesome test ;)

**Comment by WeiiswurstDev** [comment]
no problem

**Pull request merged**